malariagen_data/anoph/h12.py

```diff
@@ -514,6 +514,305 @@ def plot_h12_gwss(
         else:
             return fig
 
+    @check_types
+    @doc(
+        summary="Plot h12 GWSS data track with multiple traces overlaid.",
+    )
+    def plot_h12_gwss_multi_overlay_track(
+        self,
+        contig: base_params.contig,
+        cohorts: base_params.cohorts,
+        window_size: h12_params.multi_window_size,
+        cohort_size: Optional[base_params.cohort_size] = h12_params.cohort_size_default,
+        sample_query: Optional[base_params.sample_query] = None,
+        analysis: hap_params.analysis = base_params.DEFAULT,
+        min_cohort_size: Optional[
+            base_params.min_cohort_size
+        ] = h12_params.min_cohort_size_default,
+        max_cohort_size: Optional[
+            base_params.max_cohort_size
+        ] = h12_params.max_cohort_size_default,
+        title: Optional[gplt_params.title] = None,
+        sample_query_options: Optional[base_params.sample_query_options] = None,
+        sample_sets: Optional[base_params.sample_sets] = None,
+        colors: gplt_params.colors = bokeh.palettes.d3["Category10"][10],
+        random_seed: base_params.random_seed = 42,
+        sizing_mode: gplt_params.sizing_mode = gplt_params.sizing_mode_default,
+        width: gplt_params.width = gplt_params.width_default,
+        height: gplt_params.height = 200,
+        show: gplt_params.show = True,
+        x_range: Optional[gplt_params.x_range] = None,
+        output_backend: gplt_params.output_backend = gplt_params.output_backend_default,
+    ) -> gplt_params.figure:
+        cohort_queries = self._setup_cohort_queries(
+            cohorts=cohorts,
+            sample_sets=sample_sets,
+            sample_query=sample_query,
+            sample_query_options=sample_query_options,
+            cohort_size=cohort_size,
+            min_cohort_size=None,
+        )
+
+        if isinstance(window_size, int):
+            window_size = {k: window_size for k in cohort_queries.keys()}
+        elif isinstance(window_size, dict):
+            if np.sort(window_size.keys()) != np.sort(cohort_queries.keys()):
+                raise ValueError("Cohorts and window_sizes should have the same keys.")
```

Comment:
Just tried this out and it's raising an error, see below. Also in cases like this probably better to test for `Mapping` rather than `dict`, as we could accept anything which supports the `Mapping` interface.

Perhaps this...

```suggestion
        elif isinstance(window_size, Mapping):
            if set(window_size.keys()) != set(cohort_queries.keys()):
                raise ValueError("Cohorts and window_sizes should have the same keys.")
```

```
    558 elif isinstance(window_size, dict):
--> 559     if np.sort(window_size.keys()) != np.sort(cohort_queries.keys()):
    560         raise ValueError("Cohorts and window_sizes should have the same keys.")
    562 # Compute H12.

File ~/.cache/pypoetry/virtualenvs/malariagen-data-XzDLJC19-py3.10/lib/python3.10/site-packages/numpy/_core/fromnumeric.py:1077, in sort(a, axis, kind, order, stable)
   1075 else:
   1076     a = asanyarray(a).copy(order="K")
-> 1077 a.sort(axis=axis, kind=kind, order=order, stable=stable)
   1078 return a

AxisError: axis -1 is out of bounds for array of dimension 0
```